ptpython/python_input.py

```diff
@@ -361,8 +361,12 @@ def get_compiler_flags(self) -> int:
         flags = 0
 
         for value in self.get_globals().values():
-            if isinstance(value, __future__._Feature):
-                flags |= value.compiler_flag
+            try:
+                if isinstance(value, __future__._Feature):
+                    f = value.compiler_flag
+                    flags |= f
+            except:
```

Comment:
Can we be more precise regarding the exception we are catching?
A bare except is usually not a good idea.

Reply:
(Also, one line of documentation in the code would be good for situations like this.)

Reply:
Ok, now I understand. We need to catch every exception indeed, because the instance-check can raise anything.

Reply:
Thanks for the quick feedback. I have added a comment in the code and put `except BaseException` to avoid the bare except